File: server_side-backend/sis-pas/src/main/java/com/example/sispas/service/ExerciseService.java
package com.example.sispas.service;

import com.example.sispas.dto.ExerciseDTO;
import com.example.sispas.model.Exercise;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface ExerciseService {
    List<ExerciseDTO> getExercisesByFacilityId(Long facilityId);
    Exercise saveExercise(Long facilityId, ExerciseDTO exerciseRequest);
    Page<ExerciseDTO> getExercisesByUserId(Long userId, Pageable pageable);
    List<ExerciseDTO> getExercisesByUserId(Long userId);
}

File: client_side-frontend/sit-pass/src/app/services/exercise.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ExerciseDTO } from '../model/exercise.model';
import { AuthService } from './auth.service';

@Injectable({
  providedIn: 'root'
})
export class ExerciseService {
  private apiUrl = 'http://localhost:8600/api/exercises';

  constructor(private http: HttpClient, private authService: AuthService) {}

  getExercisesByFacilityId(facilityId: number): Observable<ExerciseDTO[]> {
    const headers = this.authService.getAuthHeaders();
    return this.http.get<ExerciseDTO[]>(`${this.apiUrl}/facility/${facilityId}`, { headers });
  }

  reserveExercise(exerciseRequest: any): Observable<ExerciseDTO> {
    const headers = this.authService.getAuthHeaders();
    return this.http.post<ExerciseDTO>(`${this.apiUrl}/reserve`, exerciseRequest, { headers });
  }

  getExercisesByUserId(userId: number, page: number = 0, size: number = 5): Observable<any> {
    const headers = this.authService.getAuthHeaders();
    return this.http.get<any>(`${this.apiUrl}/user/${userId}?page=${page}&size=${size}`, { headers });
  }

  getAllExercisesByUserId(userId: number): Observable<ExerciseDTO[]> {
    const headers = this.authService.getAuthHeaders();
    return this.http.get<ExerciseDTO[]>(`${this.apiUrl}/user/${userId}/all`, { headers });
  }

}
